test/integration_test.js

```diff
@@ -766,26 +846,35 @@ describe("Integration tests", function() {
       describe("Outgoing conflict", () => {
         let syncResult;
 
-        beforeEach(() => {
+        function setupConflict(collection) {
           let recordId;
-          return fetch(`${TEST_KINTO_SERVER}/buckets/default/collections/tasks/records`, {
-            method: "POST",
-            headers: {
-              "Accept":        "application/json",
-              "Content-Type":  "application/json",
-              "Authorization": "Basic " + btoa("user:pass"),
-            },
-            body: JSON.stringify({data: {title: "task1-remote", done: true}})
+          const record = {title: "task1-remote", done: true};
+          // Ensure that the remote record looks like something that's
+          // been transformed
+          return collection._encodeRecord("remote", record).then(record => {
+            return fetch(`${TEST_KINTO_SERVER}/buckets/default/collections/${collection._name}/records`, {
+              method: "POST",
+              headers: {
+                "Accept":        "application/json",
+                "Content-Type":  "application/json",
+                "Authorization": "Basic " + btoa("user:pass"),
+              },
+              body: JSON.stringify({data: record})
+            });
           })
-            .then(_ => tasks.sync())
+            .then(_ => collection.sync())
             .then(res => {
               recordId = res.created[0].id;
-              return tasks.delete(recordId, { virtual: false });
-            }).then(_ => tasks.create({
+              return collection.delete(recordId, { virtual: false });
+            }).then(_ => collection.create({
               id: recordId,
               title: "task1-local",
               done: false
             }, { useRecordId: true }));
+        }
+
+        beforeEach(() => {
+          return setupConflict(tasks).then(() => setupConflict(tasksTransformed));
```

Comment:
In general I'm finding promise-based code more readable when `.then` calls are on their own line/block:

```js
return setupConflict(tasks)
  .then(() => setupConflict(tasksTransformed));
```

What do you think?

Note: we'll really have to consider movig to generators at some point:

```js
yield setupConflict(tasks)
yield setupConflict(tasksTransformed);
```

Later.